src/EnergyPlus/ScheduleManager.cc

```diff
@@ -2880,101 +2848,53 @@ namespace ScheduleManager {
         // na
 
         // FUNCTION LOCAL VARIABLE DECLARATIONS:
-        int WeekSchedulePointer;
-        int DaySchedulePointer;
-        int WhichHour;
-        int WhichTimeStep;
 
-        if (!ScheduleInputProcessed) {
-            ProcessScheduleInput(OutputFiles::getSingleton());
-            ScheduleInputProcessed = true;
+
+        if (ThisHour > 24) {
+            ShowFatalError("LookUpScheduleValue called with thisHour=" + fmt::to_string(ThisHour));
         }
 
         if (ScheduleIndex == -1) {
-            LookUpScheduleValue = 1.0;
-            return LookUpScheduleValue;
+            return 1.0;
         } else if (ScheduleIndex == 0) {
-            LookUpScheduleValue = 0.0;
-            return LookUpScheduleValue;
+            return 0.0;
         }
 
-        if (ThisHour < 0) { // ThisHour unspecified
-            LookUpScheduleValue = GetCurrentScheduleValue(ScheduleIndex);
-
-            //  ELSEIF (ThisHour == 0) THEN  ! odd answers when thishour=0 (initialization of shadowing)
-            //    LookUpScheduleValue=GetCurrentScheduleValue(ScheduleIndex)
-
-        } else { // ThisHour specified
-            //  so, current date, but maybe TimeStep added
+        if (!ScheduleInputProcessed) {
+            ProcessScheduleInput(OutputFiles::getSingleton());
+            ScheduleInputProcessed = true;
+        }
 
-            // Determine which Week Schedule is used
-            //  Cant use stored day of year because of leap year inconsistency
-            WeekSchedulePointer = Schedule(ScheduleIndex).WeekSchedulePointer(DayOfYear_Schedule);
+        //  so, current date, but maybe TimeStep added
 
-            // Now, which day?
-            if (DayOfWeek <= 7 && HolidayIndex > 0) {
-                DaySchedulePointer = WeekSchedule(WeekSchedulePointer).DaySchedulePointer(7 + HolidayIndex);
-            } else {
-                DaySchedulePointer = WeekSchedule(WeekSchedulePointer).DaySchedulePointer(DayOfWeek);
-            }
+        // Hourly Value
+        int thisHour = ThisHour + DSTIndicator;
+        int thisDayOfYear = DataEnvironment::DayOfYear_Schedule;
+        int thisDayOfWeek = DataEnvironment::DayOfWeek;
+        int thisHolidayIndex = DataEnvironment::HolidayIndex;
+        if (thisHour > 24) { // In case HourOfDay is 24 and DSTIndicator is 1, you're actually the next day
+            thisDayOfYear += 1;
+            thisHour -= 24;
+            thisDayOfWeek = DataEnvironment::DayOfWeekTomorrow;
+            thisHolidayIndex = DataEnvironment::HolidayIndexTomorrow;
```

Comment:
This is new by the way. But I think it makes sense... If you're looking at tomorrow, then update the DayOfWeek and the HolidayIndex accordingly.

Reply:
```
            thisDayOfWeek = DataEnvironment::DayOfWeekTomorrow;
            thisHolidayIndex = DataEnvironment::HolidayIndexTomorrow;
```

This is the cause of the difference in the `_5ZoneAirCooled_LeapYear_annual.idf`